pi5/board.go

```diff
@@ -159,19 +149,108 @@ func (b *pinctrlpi5) Reconfigure(
 		return err
 	}
 
-	b.mu.Lock()
-	defer b.mu.Unlock()
+	// make sure every pin has a name. We already know every pin has a pin
+	for _, c := range newConf.Pins {
+		if c.Name == "" {
+			c.Name = c.Pin
+		}
+	}
+
+	if err := b.addUserDefinedNames(newConf); err != nil {
+		return err
+	}
 
 	if err := b.reconfigurePullUpPullDowns(newConf); err != nil {
 		return err
 	}
+	if err := b.reconfigureInterrupts(newConf); err != nil {
+		return err
+	}
+	b.pinConfigs = newConf.Pins
+
+	return nil
+}
+
+// reconfigureInterrupts reconfigures the digital interrupts based on the new configuration provided.
+// It reuses existing interrupts when possible and creates new ones if necessary.
+func (b *pinctrlpi5) reconfigureInterrupts(newConf *rpiutils.Config) error {
+	// look at previous interrupt config, and see if we removed any
+	for _, oldConfig := range b.pinConfigs {
+		if oldConfig.Type != rpiutils.PinInterrupt {
+			continue
+		}
+		sameInterrupt := false
+		for _, newConfig := range newConf.Pins {
+			if newConfig.Type != rpiutils.PinInterrupt {
+				continue
+			}
+			// check if we still have this interrupt
+			if oldConfig.Name == newConfig.Name && oldConfig.Pin == newConfig.Pin {
+				sameInterrupt = true
+				break
+			}
+		}
+		// if we still have the interrupt, don't modify it
+		if sameInterrupt {
+			continue
+		}
+		// we no longer want this interrupt, so we will remove it and add back the pin's gpio functionality
+		bcom, ok := rpiutils.BroadcomPinFromHardwareLabel(oldConfig.Pin)
+		if !ok {
+			return errors.Errorf("cannot find GPIO for unknown pin: %s", oldConfig.Name)
+		}
+		// this actually removes the interrupt
+		interrupt, ok := b.interrupts[bcom]
+		if ok {
+			if err := interrupt.Close(); err != nil {
+				return err
+			}
+			delete(b.interrupts, bcom)
+		}
+
+		// add back the gpio pin to make it available to the user
+		b.gpios[bcom] = b.boardPinCtrl.CreateGpioPin(b.gpioMappings[oldConfig.Pin], rpiutils.DefaultPWMFreqHz)
```

Comment:
worth mentioning, if a pin gets set as an interrupt via DigitalInterruptByName, you cannot change it back to a GPIO output pin unless you restart the module :)

Reply:
theres two methods to configure interrupts using this module

1. use a script/call DigitalInterruptByName. 
2. declare a pin as an interrupt via the config. 

I cant think of a good way to handle 1 without some other way to declare a pin in interrupt mode

Reply:
good to note, whats the current behavior if you call GPIOPinbyName on a digital interrupt pin? does it error

Reply:
it does not error, the function will return the interrupt with only inputs implemented